src/Datadog.Trace.ClrProfiler.Managed/Emit/DynamicMethodBuilder.cs

```diff
@@ -158,36 +157,37 @@ internal static class DynamicMethodBuilder<TDelegate>
                                          .ToArray();
             }
 
-            Emit<TDelegate> dynamicMethod = Emit<TDelegate>.NewDynamicMethod(methodInfo.Name);
+            DynamicMethod dynamicMethod = new DynamicMethod(methodInfo.Name, returnType, parameterTypes, ObjectExtensions.Module, skipVisibility: true);
+            ILGenerator il = dynamicMethod.GetILGenerator();
 
             // load each argument and cast or unbox as necessary
             for (ushort argumentIndex = 0; argumentIndex < parameterTypes.Length; argumentIndex++)
             {
                 Type delegateParameterType = parameterTypes[argumentIndex];
                 Type underlyingParameterType = effectiveParameterTypes[argumentIndex];
 
-                dynamicMethod.LoadArgument(argumentIndex);
+                il.Emit(OpCodes.Ldarg, argumentIndex);
```

Comment:
this implementation misses the shortcuts from `Ldarg_0` to `Ldarg_3` and the short version `Ldarg_S`. I don't know if there is a performance impact by using the int16 version directly.

Reply:
Thanks for spotting this! In 0ce85d3fe2582b68df47c0e74f64d295476ccf14 I put together a small switch statement. Is that fine with you?

Reply:
I can't see 0ce85d3fe2582b68df47c0e74f64d295476ccf14 but yeah a switch case is the right thing to do here 😊

Reply:
Oof, sorry about that. I just pushed my branch up